tools/docs/tfa_doctest.py

```diff
@@ -0,0 +1,184 @@
+import doctest
+
+FLAGS = flags.FLAGS
+
+flags.DEFINE_string('module', None, 'A specific module to run doctest on.')
+flags.DEFINE_boolean('list', None,
+                     'List all the modules in the core package imported.')
+flags.DEFINE_string('file', None, 'A specific file to run doctest on.')
+
+flags.mark_flags_as_mutual_exclusive(['module', 'file'])
+flags.mark_flags_as_mutual_exclusive(['list', 'file'])
+
+PACKAGE = 'tensorflow_addons.'
+
+
+def find_modules():
+    """Finds all the modules in the core package imported.
+
+    Returns:
+      A list containing all the modules in tensorflow.python.
+    """
+
+    tf_modules = []
+    for name, module in sys.modules.items():
+        if name.startswith(PACKAGE):
+            tf_modules.append(module)
+
+    return tf_modules
+
+
+def filter_on_submodules(all_modules, submodule):
+    """Filters all the modules based on the module flag.
+
+    The module flag has to be relative to the core package imported.
+    For example, if `submodule=keras.layers` then, this function will return
+    all the modules in the submodule.
+    Args:
+      all_modules: All the modules in the core package.
+      submodule: Submodule to filter from all the modules.
+    Returns:
+      All the modules in the submodule.
+    """
+
+    filtered_modules = [
+        mod for mod in all_modules if PACKAGE + submodule in mod.__name__
+    ]
+    return filtered_modules
+
+
+def get_module_and_inject_docstring(file_path):
+    """Replaces the docstring of the module with the changed file's content.
+
+    Args:
+      file_path: Path to the file
+    Returns:
+      A list containing the module changed by the file.
+    """
+
+    file_path = os.path.abspath(file_path)
+    mod_index = file_path.find(PACKAGE.replace('.', os.sep))
+    file_mod_name, _ = os.path.splitext(file_path[mod_index:])
+    file_module = sys.modules[file_mod_name.replace(os.sep, '.')]
+
+    with open(file_path, 'r') as f:
+        content = f.read()
+
+    file_module.__doc__ = content
+
+    return [file_module]
+
+
+class TfTestCase(tf.test.TestCase):
+    def set_up(self, test):
+        self.setUp()
+
+    def tear_down(self, test):
+        self.tearDown()
+
+
+class CustomOutputChecker(doctest.OutputChecker):
+    """Changes the `want` and `got` strings.
+
+    This allows it to be customized before they are compared.
+    """
+
+    ADDRESS_RE = re.compile(r'\bat 0x[0-9a-f]*?>')
+
+    def check_output(self, want, got, optionflags):
+        # Replace python's addresses with ellipsis (`...`) since it can change on
+        # each execution.
+        want = self.ADDRESS_RE.sub('at ...>', want)
+        return doctest.OutputChecker.check_output(self, want, got, optionflags)
+
+    _MESSAGE = textwrap.dedent("""\n
+        #############################################################
+        Check the documentation
+        (https://www.tensorflow.org/community/contribute/docs_ref) on how to write testable docstrings.
```

Comment:
Should we use the same [link](https://www.tensorflow.org/community/contribute/docs_ref) or create a new one at somewhere?

Reply:
Use the same one is fine :-)